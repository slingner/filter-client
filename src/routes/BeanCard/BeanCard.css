.Bean {
  display: flex;
  flex-direction: column;
  list-style-type: none;
  font-family: 'Montserrat', sans-serif;
  color: white;
  background-color: rgb(130, 114, 93);
  border: 1px solid grey;
  box-shadow: 14px 17px 23px -14px rgba(145, 132, 116, 1);
  border-radius: 5px;
  margin: 30px;
  padding: 15px;
  /* text-align: left; */
  min-width: 300px;
  min-height: 400px;
}

.save {
  margin-top: auto;
  padding: 10px 20px 10px 20px;
  font-size: 16px;
  background-color: rgb(110, 147, 115);
  color: white;
  font-family: 'Montserrat', sans-serif;
  border: none;
}

.added {
  margin-top: auto;
  margin-bottom: -2px;
  padding: 10px 20px 10px 20px;
  text-align: center;
  border: 1px solid white;
  font-size: 16px;
}
