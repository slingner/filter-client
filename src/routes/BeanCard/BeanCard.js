import React from 'react'
import PropTypes from 'prop-types'
import './BeanCard.css'
import config from '../../config';
import TokenService from '../../services/token-service';

export default class BeanPage extends React.Component {

  postBeanIdOnUserTable = (beanId) => {
    return fetch(`${config.API_ENDPOINT}/beans`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
      body: JSON.stringify({
        coffee_bean_id: beanId
      }),
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .catch(error => {
        console.error(error)
        this.setState({ error })
      })   
  }


  handleSaveCardClick = (event, id) => {
    event.preventDefault()
    this.postBeanIdOnUserTable(id)
      console.log(id)
  }

  render() {
    return (
          <div className='Bean'>
            <h3 className='Bean_name'>
             {this.props.bean_name}
            </h3>
            <p className='Bean_origin'>
             Origin: {this.props.bean_origin}
            </p>
            <p className='Bean_masl'>
             Meters Above Sea Level: {this.props.bean_masl}
            </p>
            <p className='Bean_grower'>
             Grower: {this.props.bean_grower}
            </p>
            <p className='Bean_process'>
             Process: {this.props.bean_process}
            </p>
            <p className='flavor_notes'>
             Flavor Notes: {this.props.flavor_notes}
            </p>
            <button className='save' onClick={(e) => this.handleSaveCardClick(e, this.props.id)}>Save</button>
          </div>
        )}
  }


BeanPage.propTypes = {
  bean_name: PropTypes.string.isRequired,
}





