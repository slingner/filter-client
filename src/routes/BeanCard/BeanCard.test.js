import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BeanCard from './BeanCard'
import BeanListContext from '../../contexts/BeansListContext';


describe(`<BeanCard />`, () => {
  //Snapshot Testing
  it('renders BeanCard by default', () => {
    const wrapper = shallow(
    <BeanListContext.Provider>
      <BeanCard />
    </BeanListContext.Provider>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})