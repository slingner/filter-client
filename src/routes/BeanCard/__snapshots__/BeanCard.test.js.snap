// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeanCard /> renders BeanCard by default 1`] = `<BeanPage />`;
